api/envoy/api/v2/core/config_source.proto

```diff
@@ -113,6 +121,18 @@ message ConfigSource {
     // When set, ADS will be used to fetch resources. The ADS API configuration
     // source in the bootstrap configuration is used.
     AggregatedConfigSource ads = 3;
+    // [#not-implemented-hide:]
+    // When set, the client will access the resources from the same server it got the
```

Comment:
I think we need to have some more discussion on this approach. Specifically, if the idea is really to make things like LRS sticky, it's arguably not the most practical approach, since it provides little guarantee. After speaking to Vishal about this, I think a better approach might be credential indirection, so let's discuss this next week.

Reply:
What kind of guarantee are you concerned about not having here?

I think that even if we decide on a different approach for LRS, this would still be useful for cases where a management server is not using ADS but wants to redirect clients to itself when linking from LDS to RDS or from CDS to EDS.  But I don't have a concrete use-case that needs that right now, so that is a bit speculative.

In any case, I'm happy to discuss this further when you get back.

Reply:
Let's just discuss in the meeting we already have scheduled for next week? While I don't have any inherent objection to the feature I am a bit concerned about implementation complexity.

Reply:
Yeah, let's talk about this when we meet next week.  I had originally scheduled the meeting about the LRS-in-CDS issue, but we now have enough open discussions that I figure we can just use that meeting to talk about all of them (or at least as many as we can fit into the available time).